lock-api/src/main/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtils.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.timelock.api.ConjureChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureCreatedChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureDeletedChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureLockRequestMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureUnchangedChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureUpdatedChangeMetadata;
+import com.palantir.conjure.java.lib.Bytes;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.watch.ChangeMetadata.Created;
+import com.palantir.lock.watch.ChangeMetadata.Deleted;
+import com.palantir.lock.watch.ChangeMetadata.Unchanged;
+import com.palantir.lock.watch.ChangeMetadata.Updated;
+import com.palantir.logsafe.Unsafe;
+import com.palantir.util.IndexEncodingUtils;
+import com.palantir.util.IndexEncodingUtils.ChecksumType;
+import com.palantir.util.IndexEncodingUtils.IndexEncodingResult;
+import com.palantir.util.IndexEncodingUtils.KeyListChecksum;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.immutables.value.Value;
+
+public final class ConjureLockRequestMetadataUtils {
+    private ConjureLockRequestMetadataUtils() {}
+
+    @VisibleForTesting
+    static final ChecksumType DEFAULT_CHECKSUM_TYPE = ChecksumType.CRC32_OF_DETERMINISTIC_HASHCODE;
+
+    public static ConjureMetadataConversionResult toConjureIndexEncoded(
+            Set<LockDescriptor> lockDescriptors, LockRequestMetadata metadata) {
+        IndexEncodingResult<LockDescriptor, ConjureChangeMetadata> encoded = IndexEncodingUtils.encode(
+                lockDescriptors,
+                metadata.lockDescriptorToChangeMetadata(),
+                changeMetadata -> changeMetadata.accept(ChangeMetadataToConjureVisitor.INSTANCE),
+                DEFAULT_CHECKSUM_TYPE);
+        KeyListChecksum checksum = encoded.keyListChecksum();
+        ConjureLockRequestMetadata conjureLockRequestMetadata = ConjureLockRequestMetadata.builder()
+                .indexToChangeMetadata(encoded.indexToValue())
+                .checksumTypeId(checksum.type().getId())
```

Comment:
Any reason why we don't just make this similar to `KeyListChecksum`, so that it's encapsulated into it's own class?

Reply:
No particular reason, just saving some space on the wire. Are you in favor of encapsulating it?

Reply:
Did the encapsulation